nl_server/gcs.py

```diff
@@ -79,10 +81,20 @@ def download_model_folder(model_folder: str) -> str:
   directory = TEMP_DIR
 
   # Only download if needed.
-  if os.path.exists(os.path.join(directory, model_folder)):
-    return os.path.join(directory, model_folder)
+  model_path = os.path.join(directory, model_folder)
+  if os.path.exists(model_path):
+    # Check if this path can still be loaded as a Sentence Transformer
```

Comment:
How long does it take?  Wonder if we can avoid doing this if we know for sure we're running on GCP?

Reply:
Not very long. I'd say may be 3-5seconds tops?

Reply:
Maybe can do this only if `_use_cache(os.environ.get('FLASK_ENV'))` is true?